### PR TITLE
Add grenade maker's bag for chemists to organize grenade components.

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Adds a grenade maker's bag that holds grenade components and beakers (but *not* grenades themselves) with a carrying capacity of 50 items.  Its sprite is a recolored chemistry bag with a cute little grenade icon. Two of them may be found in the Chemdrobe for $50 each.

This pull also spawns two grenadier belts as obtainable items on the station: one in the Chemdrobe ($125) and one in the Security vending machine ($150, default premium price), both as premium items.  

As a final tweak, I added a third chemistry bag to the Chemdrobe in case a replacement is needed (chemist goes missing, new chemist gets hired, assistant wants to do ghetto chemistry, etc.).

## Why It's Good For The Game

Chemistry is a deep and complex system with broad applications, and yet we see only a tiny slice of it each round, with medicine in high demand and other personnel such as medical doctors often filling in for the two overworked medbay chemists (and that's assuming both of them are alive and on duty all round).  I want to add a tool that will make non-medical chemistry a little bit more accessible so players can dive deeper into this satisfying game mechanic on a regular basis.

This pull gives chemists (or traitors) the ability to organize their grenade components in a sensible bag without eating all of their inventory. 

I wanted to encourage more people to take part in the full business of what grenade and chemical mixing is capable of--to help them tinker more with making grenades and other signalled devices.  This grenade maker's bag will find a use in many scenarios.  It will be appreciated by chemists specializing in grenade production, clowns who want to make glitter bombs, greytiders who want to empty out the Toxins workbench, and firefighters who want atmos foam grenades. 

With these changes, as well as an extra chemistry bag thrown into the Chemdrobe for good measure, I hope we can see players performing chemical services for all parts of the station--there are many untapped opportunities to apply chemistry to mining, security, engineering, and other pursuits, and I want to help players unlock that potential.

## Changelog
:cl:
add: Add a grenade maker's bag for chemistry so players can organize all their grenade components.  Spawns in one empty grenadier belt in both the Chemdrobe and Security vendors.  Chemdrobe gets an extra chemistry bag. 
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
are we gonna make bags normal sized items as they are pill bottle++, infinte plant storage that fits in your box++

**Comment by Barhandar** [comment]
>almost exactly the reason separated chemicals was removed from botany (that is, hundreds of grenade plants that fit in your pocket via plant bag)

>HURR LETS ADD SAME THING FOR ACTUAL GRENADES

**Comment by Eskjjlj** [comment]
The intent is good but when adding features you need to think throughly about balance.

Just imagine a chemist with 90 hellfoam grenades in his pocket.

**Comment by kriskog** [comment]
Why shouldn't grenades take up bag space? They can be extremely powerful, and are small items. Surely you can spare some room in a box for them?
There is also a grenade belt you could modify for this purpose: /obj/item/storage/belt/grenade

**Review by 81Denton — CHANGES_REQUESTED** [review]
I totally get the sentiment, managing 50 igniters, cable coils, voice analyzers etc. in your inventory is interface hell. Being able to shove up to 90 grenades into a single bag sounds like it'll cause balance issues though.

I'd rather see this issue solved in one of two (or even both) ways: 
1) An assembly/electronics/whatever bag that holds igniters, signalers, prox sensors and all the other extremely annoying small items that clutter up your inventory and nearby tables. 
No grenades or grenade assemblies though, please.
This could then spawn in chemical closets: `/obj/structure/closet/secure_closet/chemical`
2) A way to obtain empty grenade belts, for example a as a premium ChemDrobe vendor item, cargo crate or botany crafting item (from leather stacks).

**Comment by anconfuzedrock** [comment]
nobody should be carrying so many grenades that they need a fucking 90 grenade bag. Anyone who needs to carry nearly that many can get a bag of holding and/or boxes. Adding this to the code just means you want to add it to the game later, we have grenadier's belts already.

**Comment by actioninja** [comment]
no

**Comment by MegaEmpirical** [comment]
It's called a grenadier belt. The nuke ops have one.

**Comment by FloranOtten** [comment]
90 grenades. Even without hellfoam grenades, imagine 90 basic potassium water grenades. You could destroy a _lot_ with that.

**Comment by kriskog** [comment]
Some kind of multi-purpose crafting bag wouldn't be a shit idea. Just make sure you add a recipe.

You also seem to have an outdated icon file, you are removing some stuff.

**Comment by knacker48** [comment]
This feels more like one of those role restricted syndicate items

**Comment by Qydra** [comment]
> I totally get the sentiment, managing 50 igniters, cable coils, voice analyzers etc. in your inventory is interface hell. Being able to shove up to 90 grenades into a single bag sounds like it'll cause balance issues though.
> 
> I'd rather see this issue solved in one of two (or even both) ways:
> 
> 1. An assembly/electronics/whatever bag that holds igniters, signalers, prox sensors and all the other extremely annoying small items that clutter up your inventory and nearby tables.
>    No grenades or grenade assemblies though, please.
>    This could then spawn in chemical closets: `/obj/structure/closet/secure_closet/chemical`
> 2. A way to obtain empty grenade belts, for example a as a premium ChemDrobe vendor item, cargo crate or botany crafting item (from leather stacks).

I've adjusted the bag's eligible contents per 81Denton's suggestion above.  I'd like to be able to store JUST chemical grenade casings and not the grenades, but I haven't figured out how to do that yet, so no grenade casings.

**Comment by MegaEmpirical** [comment]
I'd suggest not allowing partially constructed grenades either, lest they be one step away from being finished as an exploit.

**Comment by Qydra** [comment]
> I'd suggest not allowing partially constructed grenades either, lest they be one step away from being finished as an exploit.

Okay, I'll keep grenade casings out then.

**Comment by Qydra** [comment]
I have just added spawn locations to the grenade maker's bag: 2 will spawn in the Chemdrobe machine for $50 each.  I have also added two grenadier belts to the Chemdrobe ($75) and one grenadier belt to the security vending machine ($150).

**Comment by optimumtact** [comment]
Changes to storage and inventory are actually usually the most balance modifying thing in the game, because of how limited most peoples inventories are.

That being said, Im not sure I see the value in this, you can put small items in boxes which can be popped out of cardboard, just add some stacks of cardboard in chemistry and a labeller so they can label their parts boxes.

**Comment by MegaEmpirical** [comment]
Giving a grenadier belt to sec makes sense though I feel like giving them to chem is a bit weird.

**Comment by Qydra** [comment]
> Giving a grenadier belt to sec makes sense though I feel like giving them to chem is a bit weird.

Eh, maybe I'll give chem only one grenadier belt and make them fight over it.  Or put all three in the sec vending machine and make the chemists seek permission to get one, although I hesitate to give Security one more thing to have to deal with every round.

**Comment by Qydra** [comment]
All done here and ready for review, I think.  You've helped me refine this idea quite a bit, and I think the end product is about ready.  I look forward to further feedback.

**Comment by anconfuzedrock** [comment]
Still a bad end product. nobody should have more than a few grenades and there's never a reason not to just store said grenades in a box or something. sec and military belts can hold grenades already, This is just weird, this is very debatably similar to if we added an oil slime extract to xenobiology that can hold 90 oil slime extracts and nothing else.

**Comment by davethwave** [comment]
Now that this only allows you to contain parts I am ok with it if it matters.

File: code/game/objects/items/storage/bags.dm [hunk 1]
```diff
@@ -346,6 +346,26 @@
...
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bag_grenade"
+	desc = "A bag for storing grenade components."
+	w_class = WEIGHT_CLASS_TINY
+	resistance_flags = FLAMMABLE
+
+/obj/item/storage/bag/grenade_parts/ComponentInitialize()
+	. = ..()
+	GET_COMPONENT(STR, /datum/component/storage)
+	STR.max_combined_w_class = 200
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/storage/bags.dm]: will you ever reach this number?

File: code/game/objects/items/storage/bags.dm [hunk 2]
```diff
@@ -346,6 +346,26 @@
...
+	STR.max_items = 50
+	STR.insert_preposition = "in"
+	STR.can_hold = typecacheof(list(/obj/item/reagent_containers/glass/beaker, /obj/item/assembly/igniter, /obj/item/assembly/health, /obj/item/assembly/infra, /obj/item/assembly/mousetrap, /obj/item/assembly/prox_sensor, /obj/item/assembly/signaler, /obj/item/assembly/timer, /obj/item/assembly/voice, /obj/item/stack/cable_coil, /obj/item/screwdriver))
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/storage/bags.dm]: if it isn't a subtype of `/obj/item/assembly` I'd like it removed.

If you have to refluff it to be an "assembly bag" then fine.

**Comment by ExcessiveUseOfCobblestone** [comment]
bags are honestly cringe but *shrug*

**Comment by PKPenguin321** [comment]
Storage creep

File: code/game/objects/items/storage/bags.dm (hunk 1 above)
Comment (Qydra) [reply code/game/objects/items/storage/bags.dm]: I don't actually know what that variable does, but it's the same as the Chemistry bag.

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/storage/bags.dm]: it's the sum of item's weights.

I think all assemblies listed are tiny (1) so at most you could ever get is 50 which is why I asked.

File: code/game/objects/items/storage/bags.dm (hunk 2 above)
Comment (Qydra) [reply code/game/objects/items/storage/bags.dm]: Does that include screwdrivers and cables?

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/storage/bags.dm]: yes

**Comment by optimumtact** [comment]
the general feedback from maintainers is negative.

**Comment by Qydra** [comment]
I think I've been dissuaded from this idea anyway.  I'll think of a better one.  Thanks for the feedback and the learning, all.

**Pull request closed without merging**